src/zhinst/toolkit/driver/devices/pqsc.py

```diff
@@ -196,42 +208,40 @@ def _check_zsync_connection(
 
         Args:
             ports: Port number to check the ZSync connection for.
-            timeout: Maximum time in seconds the program waits (default: 30).
-            sleep_time: Time in seconds to wait between requesting the status
-                (default: 1)
+            timeout: Maximum time in seconds the program waits.
+            sleep_time: Time in seconds to wait between requesting the status.
 
         Raises:
             TimeoutError: If the process of establishing a ZSync connection the
                 specified port exceeds the specified timeout.
         """
         status_node = self.zsyncs[port].connection.status
-        start_time = time.time()
         try:
-            status_node.wait_for_state_change(
-                0, invert=True, timeout=timeout, sleep_time=sleep_time
-            )
-            status_node.wait_for_state_change(
-                1,
-                invert=True,
-                timeout=max(0, timeout - (time.time() - start_time)),
-                sleep_time=sleep_time,
-            )
+            # Waits until the status node is "connected" (2)
+            status_node.wait_for_state_change(2, timeout=timeout, sleep_time=sleep_time)
         except TimeoutError as error:
             raise TimeoutError(
                 "Timeout while establishing ZSync connection to the instrument "
                 f"on the port {port}."
             ) from error
         return status_node() == 2
 
-    def find_zsync_worker_port(self, device: BaseInstrument) -> int:
+    def find_zsync_worker_port(
+        self,
+        device: BaseInstrument,
+        timeout: float = 10,
```

Comment:
The default value must match the typehint.
```suggestion
        timeout: float = 10.0,
```

Reply:
done.